### ISSUE TITLE
[clang-tidy]: clang-diagnostic-error can't find system header files

### ISSUE THREAD

**Issue by igagis** [llvm/llvm-project#59738]
# To reproduce

create sample `main.cpp` file:
```cpp
#include <memory>

int main(int argc, const char** argv){
    return 0;
}
```

run `clang-tidy`:
```
clang-tidy main.cpp
```

# Expected result
no errors found by `clang-tidy`

# Actual result
following error is reported by `clang-tidy`:
```
1 error generated.
Error while processing /home/ivan/prj/test/main.cpp.
/home/ivan/prj/test/main.cpp:1:10: error: 'memory' file not found [clang-diagnostic-error]
#include <memory>
         ^~~~~~~~
Found compiler error(s)
```

# version
```console
$ clang-tidy --version
Ubuntu LLVM version 16.0.0
  Optimized build.
```

**Comment by EugeneZelenko** [comment]
Does `clang` without additional command-line arguments works?

**Comment by igagis** [comment]
nope:
```console
$ clang main.cpp 
main.cpp:1:10: fatal error: 'memory' file not found
#include <memory>
         ^~~~~~~~
1 error generated.
```

but `gcc` works.

**Comment by junaire** [comment]
It should be `clang++`, not `clang`.

Also, can you try `clang++ -### main.cpp` to see if clang searches the correct system header path?

**Comment by igagis** [comment]
```console
$ clang++ main.cpp 
main.cpp:1:10: fatal error: 'memory' file not found
#include <memory>
         ^~~~~~~~
1 error generated.
```

```console
$ clang++ -### main.cpp
Ubuntu clang version 16.0.0 (++20221229093801+1f84e72b7ba0-1~exp1~20221229093909.650)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
 "/usr/lib/llvm-16/bin/clang" "-cc1" "-triple" "x86_64-pc-linux-gnu" "-emit-obj" "-mrelax-all" "--mrelax-relocations" "-disable-free" "-clear-ast-before-backend" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "main.cpp" "-mrelocation-model" "pic" "-pic-level" "2" "-pic-is-pie" "-mframe-pointer=all" "-fmath-errno" "-ffp-contract=on" "-fno-rounding-math" "-mconstructor-aliases" "-funwind-tables=2" "-target-cpu" "x86-64" "-tune-cpu" "generic" "-mllvm" "-treat-scalable-fixed-error-as-warning" "-debugger-tuning=gdb" "-fcoverage-compilation-dir=/home/ivan/prj/test" "-resource-dir" "/usr/lib/llvm-16/lib/clang/16" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/x86_64-linux-gnu" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/backward" "-internal-isystem" "/usr/lib/llvm-16/lib/clang/16/include" "-internal-isystem" "/usr/local/include" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../x86_64-linux-gnu/include" "-internal-externc-isystem" "/usr/include/x86_64-linux-gnu" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-fdeprecated-macro" "-fdebug-compilation-dir=/home/ivan/prj/test" "-ferror-limit" "19" "-fgnuc-version=4.2.1" "-fcxx-exceptions" "-fexceptions" "-fcolor-diagnostics" "-faddrsig" "-D__GCC_HAVE_DWARF2_CFI_ASM=1" "-o" "/tmp/main-489b5a.o" "-x" "c++" "main.cpp"
 "/usr/bin/ld" "-pie" "-z" "relro" "--hash-style=gnu" "--build-id" "--eh-frame-hdr" "-m" "elf_x86_64" "-dynamic-linker" "/lib64/ld-linux-x86-64.so.2" "-o" "a.out" "/lib/x86_64-linux-gnu/Scrt1.o" "/lib/x86_64-linux-gnu/crti.o" "/usr/bin/../lib/gcc/x86_64-linux-gnu/12/crtbeginS.o" "-L/usr/bin/../lib/gcc/x86_64-linux-gnu/12" "-L/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../lib64" "-L/lib/x86_64-linux-gnu" "-L/lib/../lib64" "-L/usr/lib/x86_64-linux-gnu" "-L/usr/lib/../lib64" "-L/lib" "-L/usr/lib" "/tmp/main-489b5a.o" "-lstdc++" "-lm" "-lgcc_s" "-lgcc" "-lc" "-lgcc_s" "-lgcc" "/usr/bin/../lib/gcc/x86_64-linux-gnu/12/crtendS.o" "/lib/x86_64-linux-gnu/crtn.o"
```

**Comment by junaire** [comment]
I have run into this similar issue before, though I forgot how I fixed that...
Can you do: `ls /usr/include/c++` to check whether you have multiple versions of libstdc++ installed?

**Comment by junaire** [comment]
Generally, the fix should be installing some packages like `libstdc++-12-dev`. I can't say it's a bug though...

**Comment by igagis** [comment]
```console
$ ls /usr/include/c++
11
ivan@igagislp:~/prj/utki$ ls /usr/include/c++/11/
algorithm  bitset    chrono     complex             cstdarg   ctime     exception     functional        iterator  memory_resource  pstl              set              stop_token    tr1            utility
any        cassert   cinttypes  complex.h           cstdbool  cuchar    execution     future            latch     mutex            queue             shared_mutex     streambuf     tr2            valarray
array      ccomplex  ciso646    concepts            cstddef   cwchar    experimental  initializer_list  limits    new              random            source_location  string        tuple          variant
atomic     cctype    climits    condition_variable  cstdint   cwctype   ext           iomanip           list      numbers          ranges            span             string_view   typeindex      vector
backward   cerrno    clocale    coroutine           cstdio    cxxabi.h  fenv.h        ios               locale    numeric          ratio             sstream          syncstream    typeinfo       version
barrier    cfenv     cmath      csetjmp             cstdlib   debug     filesystem    iosfwd            map       optional         regex             stack            system_error  type_traits
bit        cfloat    codecvt    csignal             cstring   decimal   forward_list  iostream          math.h    ostream          scoped_allocator  stdexcept        tgmath.h      unordered_map
bits       charconv  compare    cstdalign           ctgmath   deque     fstream       istream           memory    parallel         semaphore         stdlib.h         thread        unordered_set
```

> Generally, the fix should be installing some packages like libstdc++-12-dev. I can't say it's a bug though...

why isn't it installed together with `clang` as dependency? I'm using Ubuntu jammy, and installing latest clang from
```
deb http://apt.llvm.org/jammy llvm-toolchain-jammy main
```

**Comment by igagis** [comment]
```
sudo apt install libstdc++-12-dev
```
fixed the problem for me, but again, why isn't it installed as `clang-tidy` dependency automatically?

**Comment by EugeneZelenko** [comment]
@igagis: There is also `libc++` that is part of LLVM.

**Comment by xgupta** [comment]
> ```
> sudo apt install libstdc++-12-dev
> ```
> 
> fixed the problem for me, but again, why isn't it installed as `clang-tidy` dependency automatically?

Ubuntu distro/package manager question, libstdc++ is GNU/GCC project, its installation can't control by llvm-project.
